src/System.Private.ServiceModel/tools/setupfiles/sync-pr.cmd

```diff
@@ -0,0 +1,43 @@
+@echo off 
+setlocal
+
+REM 
+REM This calls a web service, specified on the command line, which syncs a 
+REM remote server's repo to the correct PR. 
+REM This script is meant to be called from a GitHub Pull Request. 
+REM If you must, you can fake it by setting the 'ghprbPullId' environment variable
+REM 
+
+set __args=%*
+
+echo.
+echo  WCF Remote Github Repo sync script
+
+if '%__args%'=='' (
+    echo   [%~n0] A URL must be specified for the pull request synchronization URL
+    echo     Usage:  %~n0 [sync url]
+    echo   sync-url - URL on remote server for PR synchronization
+    echo.
+    echo   Example:  %~n0 http://wcfci-sync-server/repo/sync.ashx
+    set __EXITCODE=1
+    goto done
+)
+
+set __SYNC_HOST_URL=%1
+
+if '%ghprbPullId%'=='' (
+    echo   [%~n0] This script should only be called only from the context of a GitHub Pull Request
```

Comment:
Sorry, I may not make my point clear. There should be only one person (caller vs. callee) who owns the knowledge of ghprbPullId env var. If you make it as a parameter, then caller owns it and the callee should fail if it is not passed in (so not optional). It's the magic that the callee still works even without parameter passed in makes the maintenance hard and error prone.

I am okay with the other way too (does not accept it as a parameter) - as you said, intentionally make it harder for anyone to use it on non-CI environment :smile:.